tests/compile-and-dump/InsertionSort/InsertionSortImp.ghc84.template

```diff
@@ -7,11 +7,12 @@ InsertionSort/InsertionSortImp.hs:(0,0)-(0,0): Splicing declarations
     instance SuppressUnusedWarnings SuccSym0 where
       suppressUnusedWarnings
         = snd ((GHC.Tuple.(,) SuccSym0KindInference) GHC.Tuple.())
-    data SuccSym0 (l :: TyFun Nat Nat)
-      = forall arg. SameKind (Apply SuccSym0 arg) (SuccSym1 arg) =>
-        SuccSym0KindInference
+    data SuccSym0 (l :: TyFun Nat Nat) :: Type
+      where
+        SuccSym0KindInference :: forall l arg.
+                                 SameKind (Apply SuccSym0 arg) (SuccSym1 arg) => SuccSym0 l
     type instance Apply SuccSym0 l = Succ l
-    data instance Sing (z :: Nat)
+    data instance Sing :: Nat -> Type :: Nat -> Type
```

Comment:
If you're wondering why GHC is stuttering here with `:: Nat -> Type :: Nat -> Type`, it's due to [GHC Trac #14817](https://ghc.haskell.org/trac/ghc/ticket/14817), which I'm looking into now.